src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostAbortedExceptionTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using Xunit;
+
+namespace Microsoft.Extensions.Hosting.Unit.Tests
+{
+    public sealed class HostAbortedExceptionTests
+    {
+        [Fact]
+        public void TestEmptyException()
+        {
+            var exception = new HostAbortedException();
+            Assert.Null(exception.InnerException);
+            Assert.Throws<HostAbortedException>(new Action(() => throw exception));
+        }
+
+        [Theory]
+        [InlineData("Host aborted.", false)]
+        [InlineData("Host aborted.", true)]
+        public void TestException(string? message, bool innerException)
+        {
+            HostAbortedException exception = innerException
+                ? new HostAbortedException(message, new Exception())
+                : new HostAbortedException(message);
+
+            Assert.Equal(message, exception.Message);
+            if (innerException)
+            {
+                Assert.NotNull(exception.InnerException);
+            } 
+            
+            HostAbortedException thrownException = Assert.Throws<HostAbortedException>(new Action(() => throw exception));
+            Assert.Equal(message, thrownException.Message);
+        }
+
+        [Fact]
+        public void TestSerialization()
+        {
+            var exception = new HostAbortedException();
+            using var serializationStream = new MemoryStream();
+
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(serializationStream, exception);
+
+            using var deserializationStream = new MemoryStream(serializationStream.ToArray());
+            HostAbortedException deserializedException = (HostAbortedException) formatter.Deserialize(deserializationStream);
+
+            Assert.Equal(exception.Message, deserializedException.Message);
```

Comment:
It might be good to set the Message, to ensure that gets serialized correctly.

Reply:
You are right. I will include the case.

Unless @bartonjs decides to remove the binary serialization after all.